include/rcutils/macros.h

```diff
@@ -64,6 +64,32 @@ extern "C"
 #define RCUTILS_STRINGIFY(x) RCUTILS_STRINGIFY_IMPL(x)
 #define RCUTILS_UNUSED(x) (void)(x)
 
+#if defined _WIN32 || defined __CYGWIN__
+#define RCUTILS_ATTRIBUTE_PRINTF_FORMAT(format_string_index, first_to_check_index)
+#else
+// This macro can be used to annotate printf-like functions which are relying
+// on a format string and a variable number of arguments.
+//
+// This enables GCC to emit warnings if dangerous patterns are detected.
+// See GCC documentation for details:
+// https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html
+//
+// format_string_index is the index of the format string passed to the function
+// first_to_check_index is the index of the first "optional argument".
+//
+// For the following function:
+// int snprintf(char *str, size_t size, const char *format, ...);
+//              ^^^^^^^^^  ^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^  ^^^
+//              ARG 1      ARG 2        ARG 3               ARG 4
+//                                      format string       first optional argument
+//
+// format_string_index value would be 3, first_to_check_index value would be 4.
+//
+// IMPORTANT: the first argument has an index of ONE (not zero!).
```

Comment:
Please use a proper docblock instead of just comment lines to document the macro (in both cases).

Reply:
Done. I changed the doc for the Windows case. Let me know if this is the right strategy.

Reply:
LGTM